### PR TITLE
Parameterized extract

### PR DESCRIPTION
**Pull request by elijahbenizzy**
[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
OK, this is ready. Calling `@parameterized_frame` experimental till we get a question on it -- the API is a bit clunky but I haven't ifgured out how to handle it perfectly yet...

**Comment by skrawcz** [comment]
@elijahbenizzy as discussed, I think making this live under experimental would be better short term.

**Review by skrawcz — APPROVED** [review]
seems reasonable -- just a few minor things. Fix them and I approve.

File: hamilton/experimental/decorators/parameterize_frame.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+import pandas as pd
+
+from hamilton.function_modifiers import base, parameterize_extract_columns, source, value
+from hamilton.function_modifiers.expanders import ParameterizedExtract
+
+
+def _get_dep_type(dep_type: str):
+    if dep_type == "out":
+        return None
+    if dep_type == "value":
```
[the remark this answers is not in the document]
Comment (elijahbenizzy) [reply hamilton/experimental/decorators/parameterize_frame.py:12]: Going to keep it with `if/return` as opposed to `elif`, but make it consistent

**Pull request merged**